modules/openy_features/openy_location/modules/openy_loc_filter/openy_loc_filter.module

```diff
@@ -106,6 +107,17 @@ function openy_loc_filter_query_repeat_schedules_block_locations_alter(Alterable
 function openy_loc_filter_query_data_wrapper_locations_alter(AlterableInterface $query) {
   $allowed_locations = \Drupal::config(LocationFilterSettingsForm::CONFIG_NAME)
     ->get('locations');
+  unset($allowed_locations['All']);
+  $changed_by_user = \Drupal::config(LocationFilterSettingsForm::CONFIG_NAME)
+    ->get('changed_by_user');
+  // If user didn't touch the settings, then display all locations by default.
+  if (!$changed_by_user) {
```

Comment:
I think no need in the `changed_by_user` setting, in the field description we have:

> All locations are not selected = All locations are selected.

So for empty `locations` there no limit by ID's. @aleevas you should check why this condition works in any case:

```php
  if (!empty($allowed_locations)) {
    $query->condition('node_field_data.nid', array_keys($allowed_locations), 'IN');
  }
```

Reply:
@hamrant 

> I think no need in the changed_by_user setting, in the field description we have:

Ok, I'll fix it

> @aleevas you should check why this condition works in any case:

That because by default we had 'All' value in **allowed_locations** array, but without any real ID's.
And this condition  ` if (!empty($allowed_locations)) ` was always correct

Reply:
So, with your
```php
unset($allowed_locations['All']);
```
It will work fine?